lib/elixir/lib/string.ex

```diff
@@ -2714,6 +2714,69 @@ defmodule String do
   @spec to_char_list(t) :: charlist
   def to_char_list(string), do: String.to_charlist(string)
 
+  @doc """
+  Returns the `string` appended by `appender`, separated by `joiner`.
+
+  ## Examples
+
+      iex> String.append("alice", "bob")
+      "alicebob"
+      iex> String.append("alice", "bob", " & ")
+      "alice & bob"
+
+  """
+  @spec append(t, t, t) :: t
+  def append(string, appender, joiner \\ "")
```

Comment:
I don't think the joiner is necessary since it can be done with two appends instead.